### PR TITLE
CODE RUB: Upgrade v2.10.0, With V2.10.1 Refactoring

### LINKED ISSUE #28: CODE RUB: Standard v2.10.0 Project Upgrade

### LINKED ISSUE #30: CODE RUB: Upgrade To Standard `v2.10.1`
Requires #29

### PR DESCRIPTION
**Pull request by glhays**
Closes: #28
Closes: #30

### CODE REVIEW

**Comment by glhays** [comment]
@cjdutoit please review.
Closed PR #25 as it became obsolete due to this upgrade to version 2.10.0

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Exceptions.Add.cs [hunk 1]
```diff
@@ -19,7 +18,7 @@ public partial class EventProcessingServiceTests
     {
         [Theory]
         [MemberData(nameof(DependencyValidationExceptions))]
-        public void ShouldThrowDependencyValidationExceptionOnAddIfDependencyValidationErrorOccurs(
+        private void ShouldThrowDependencyValidationExceptionOnAddIfDependencyValidationErrorOccurs(
```
Comment (cjdutoit) [inline LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Exceptions.Add.cs:21]: Test methods must be public. The xUnit framework identifies test methods based on their public access modifier.

```suggestion
        public void ShouldThrowDependencyValidationExceptionOnAddIfDependencyValidationErrorOccurs(
```

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Exceptions.Add.cs [hunk 2]
```diff
@@ -61,7 +61,7 @@ public partial class EventProcessingServiceTests
 
         [Theory]
         [MemberData(nameof(DependencyExceptions))]
-        public void ShouldThrowDependencyExceptionOnAddIfDependencyErrorOccurs(
+        private void ShouldThrowDependencyExceptionOnAddIfDependencyErrorOccurs(
```
Comment (cjdutoit) [inline LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Exceptions.Add.cs:64]:
```suggestion
        public void ShouldThrowDependencyExceptionOnAddIfDependencyErrorOccurs(
```

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Exceptions.Add.cs [hunk 3]
```diff
@@ -102,7 +103,7 @@ public partial class EventProcessingServiceTests
         }
 
         [Fact]
-        public void ShouldThrowServiceExceptionOnAddIfServiceErrorOcurrs()
+        private void ShouldThrowServiceExceptionOnAddIfServiceErrorOcurrs()
```
Comment (cjdutoit) [inline LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Exceptions.Add.cs:106]:
```suggestion
        public void ShouldThrowServiceExceptionOnAddIfServiceErrorOcurrs()
```

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Exceptions.Publish.cs [hunk 4]
```diff
@@ -15,19 +14,21 @@ namespace LeVent.Tests.Unit.Services.Foundations.Events
     public partial class EventProcessingServiceTests
     {
         [Fact]
-        public async Task ShouldThrowServiceExceptionOnPublishIfServiceErrorOccursAsync()
+        private async Task ShouldThrowServiceExceptionOnPublishIfServiceErrorOccursAsync()
```
Comment (cjdutoit) [inline LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Exceptions.Publish.cs:17]:
```suggestion
        public async Task ShouldThrowServiceExceptionOnPublishIfServiceErrorOccursAsync()
```

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Logic.Add.cs [hunk 5]
```diff
@@ -14,7 +13,7 @@ namespace LeVent.Tests.Unit.Services.Foundations.Events
     public partial class EventProcessingServiceTests
     {
         [Fact]
-        public void ShouldRegisterEventHandlerWithoutEventName()
+        private void ShouldRegisterEventHandlerWithoutEventName()
```
Comment (cjdutoit) [inline LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Logic.Add.cs:16]:
```suggestion
        public void ShouldRegisterEventHandlerWithoutEventName()
```

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Logic.Add.cs [hunk 6]
```diff
@@ -44,7 +43,7 @@ public void ShouldRegisterEventHandlerWithoutEventName()
         }
 
         [Fact]
-        public void ShouldRegisterEventHandlerWithEventName()
+        private void ShouldRegisterEventHandlerWithEventName()
```
Comment (cjdutoit) [inline LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Logic.Add.cs:46]:
```suggestion
        public void ShouldRegisterEventHandlerWithEventName()
```

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Logic.Publish.cs [hunk 7]
```diff
@@ -16,7 +15,7 @@ namespace LeVent.Tests.Unit.Services.Foundations.Events
     public partial class EventProcessingServiceTests
     {
         [Fact]
-        public async Task ShouldPublishEventWithoutEventNameAsync()
+        private async Task ShouldPublishEventWithoutEventNameAsync()
```
Comment (cjdutoit) [inline LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Logic.Publish.cs:18]:
```suggestion
        public async Task ShouldPublishEventWithoutEventNameAsync()
```

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Logic.Publish.cs [hunk 8]
```diff
@@ -52,7 +51,7 @@ public async Task ShouldPublishEventWithoutEventNameAsync()
         }
 
         [Fact]
-        public async Task ShouldPublishEventWithEventNameAsync()
+        private async Task ShouldPublishEventWithEventNameAsync()
```
Comment (cjdutoit) [inline LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Logic.Publish.cs:54]:
```suggestion
        public async Task ShouldPublishEventWithEventNameAsync()
```

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Validations.Add.cs [hunk 9]
```diff
@@ -16,17 +15,19 @@ namespace LeVent.Tests.Unit.Services.Foundations.Events
     public partial class EventProcessingServiceTests
     {
         [Fact]
-        public void ShouldThrowValidationExceptionOnAddIfEventHandlerIsNull()
+        private void ShouldThrowValidationExceptionOnAddIfEventHandlerIsNull()
```
Comment (cjdutoit) [inline LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Validations.Add.cs:18]:
```suggestion
        public void ShouldThrowValidationExceptionOnAddIfEventHandlerIsNull()
```

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Validations.Publish.cs [hunk 10]
```diff
@@ -14,17 +13,19 @@ namespace LeVent.Tests.Unit.Services.Foundations.Events
     public partial class EventProcessingServiceTests
     {
         [Fact]
-        public async Task ShouldThrowValidationExceptionOnAddIfEventIsNullAsync()
+        private async Task ShouldThrowValidationExceptionOnAddIfEventIsNullAsync()
```
Comment (cjdutoit) [inline LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Validations.Publish.cs:16]:
```suggestion
        public async Task ShouldThrowValidationExceptionOnAddIfEventIsNullAsync()
```

**Review by cjdutoit — CHANGES_REQUESTED** [review]
All test methods must be public. The xUnit framework identifies test methods based on their public access modifier.

File: LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Exceptions.Add.cs (hunk 1 above)
Comment (glhays) [reply LeVent.Tests.Unit/Services/Processings/Events/EventProcessingServiceTests.Exceptions.Add.cs:21]: @cjdutoit 
In and around the time of this video (link below) regarding the Standard version upgrade of 2.10.0 and as well subsequent videos prior to or post of the video. It was stated that the XUnit test methods can be made private. Although the clsss must remain public.

I did not arbitrarily decide to make them private on my own accord, based on my opinion. But to share my opinion marking them private has no ill effect on running the tests. Please inform on why they MUST be public. I did some research and found some threads stating the same as I have above and as was in the videos by Hassan.

https://youtu.be/oyJvszELGwE?si=NlbZMYYqhsgd5Jay
https://github.com/xunit/xunit/discussions/2438
https://stackoverflow.com/questions/16214684/why-is-the-xunit-runner-not-finding-my-tests/16214704#16214704
https://xunit.net/xunit.analyzers/rules/

**Comment by glhays** [comment]
> All test methods must be public. The xUnit framework identifies test methods based on their public access modifier.

In and around the time of this video (link below) regarding the Standard version upgrade of 2.10.0 and as well subsequent videos prior to or post of the video. It was stated that the XUnit test methods can be made private. Although the clsss must remain public.

I did not arbitrarily decide to make them private on my own accord, based on my opinion. But to share my opinion marking them private has no ill effect on running the tests. Please inform on why they MUST be public. I did some research and found some threads stating the same as I have above and as was in the videos by Hassan.

https://youtu.be/oyJvszELGwE?si=NlbZMYYqhsgd5Jay
https://github.com/xunit/xunit/discussions/2438
https://stackoverflow.com/questions/16214684/why-is-the-xunit-runner-not-finding-my-tests/16214704#16214704
https://xunit.net/xunit.analyzers/rules/

**Review by cjdutoit — DISMISSED** [review]

**Comment by glhays** [comment]
@cjdutoit marked as WIP need to convert from 2.10.0 -> 2.10.1 will work on as time permits.

File: LeVent/Clients/LeventClient.cs [hunk 11]
```diff
@@ -39,23 +39,27 @@ public async ValueTask PublishEventAsync(T @event, string eventName = null)
             }
             catch (EventProcessingValidationException eventProcessingValidationException)
             {
-                throw new LeVentValidationException(
-                    eventProcessingValidationException.InnerException as Xeption);
+                throw CreateLeVentValidationException(eventProcessingValidationException);
+                //throw new LeVentValidationException(
```
Comment (cjdutoit) [inline LeVent/Clients/LeventClient.cs]: Remove comments

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**